### PR TITLE
Add user's pending approval managing REST API

### PR DESCRIPTION
**Pull request by ayshsandu**
## Purpose
- Add implementation of user's pending approval managing REST API
- Swagger: https://app.swaggerhub.com/apis/ayshsandu/approvals/1.0.0#/

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/gen/java/org/wso2/carbon/identity/rest/api/user/approval/v1/MeApi.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import javax.ws.rs.core.Response;
+import javax.ws.rs.*;
+
+@Path("/me")
+
+
+@io.swagger.annotations.Api(value = "/me", description = "the me API")
+public class MeApi  {
+
+   private final MeApiService delegate = MeApiServiceFactory.getMeApi();
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/gen/java/org/wso2/carbon/identity/rest/api/user/approval/v1/MeApi.java]: @Autowired?

Comment (ayshsandu) [reply components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/gen/java/org/wso2/carbon/identity/rest/api/user/approval/v1/MeApi.java]: Added in 9c7a2de6ec4ce3b19e5e0126858a3924c6a4b7c8

File: components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.api.user.approval.common/pom.xml [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.user.api</groupId>
+            <artifactId>org.wso2.carbon.identity.api.user.common</artifactId>
+            <scope>provided</scope>
+            <version>${project.version}</version>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.api.user.approval.common/pom.xml]: New line at the end of the file is missing

File: components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/approval/v1/core/model/TaskParam.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Task Param model
+ */
+@XmlRootElement(name = "xsd-complex-type-wrapper")
+public class TaskParam {
+
+
+//    private Map<String ,String > params;
```
Comment (IsuraD) [inline components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/approval/v1/core/model/TaskParam.java]: remove commented line if not require

Comment (ayshsandu) [reply components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.rest.api.user.approval.v1/src/main/java/org/wso2/carbon/identity/rest/api/user/approval/v1/core/model/TaskParam.java]: fixed

File: components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.api.user.approval.common/pom.xml (hunk 2 above)
Comment (ayshsandu) [reply components/org.wso2.carbon.identity.api.user.approval/org.wso2.carbon.identity.api.user.approval.common/pom.xml]: fixed

**Pull request merged**